File: backend/app/schemas/medical/hospital.py
from pydantic import BaseModel, EmailStr, HttpUrl, Field
from typing import Optional, List
from datetime import datetime

class HospitalBase(BaseModel):
    name: str
    category: Optional[str] = None
    address: str
    latitude: Optional[float] = None
    longitude: Optional[float] = None
    available_beds: int = 0
    icu_beds: int = 0
    emergency_contact: str
    phone_number: Optional[str] = None
    email: Optional[EmailStr] = None
    website: Optional[HttpUrl] = None
    google_maps_link: Optional[HttpUrl] = None

class HospitalCreate(HospitalBase):
    pass

class HospitalUpdate(BaseModel):
    name: Optional[str] = None
    category: Optional[str] = None
    address: Optional[str] = None
    latitude: Optional[float] = None
    longitude: Optional[float] = None
    available_beds: Optional[int] = None
    icu_beds: Optional[int] = None
    emergency_contact: Optional[str] = None
    phone_number: Optional[str] = None
    email: Optional[EmailStr] = None
    website: Optional[str] = None
    google_maps_link: Optional[str] = None

class HospitalResponse(HospitalBase):
    id: int
    is_active: bool
    distance: Optional[float] = None  # in km, calculated if lat/lon provided
    rating: Optional[float] = 0.0
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True

File: frontend/educonnect/lib/models/hospital.dart
class Hospital {
  final int id;
  final String name;
  final String category;
  final String address;
  final int availableBeds;
  final int icuBeds;
  final String emergencyContact;
  final String? phoneNumber;
  final String? email;
  final String? website;
  final String? googleMapsLink;
  final double rating;
  final double? distance;
  final String? image;

  Hospital({
    required this.id,
    required this.name,
    required this.category,
    required this.address,
    required this.availableBeds,
    required this.icuBeds,
    required this.emergencyContact,
    this.phoneNumber,
    this.email,
    this.website,
    this.googleMapsLink,
    required this.rating,
    this.distance,
    this.image,
  });

  factory Hospital.fromJson(Map<String, dynamic> json) {
    return Hospital(
      id: json['id'],
      name: json['name'],
      category: json['category'] ?? 'General',
      address: json['address'] ?? 'No address',
      availableBeds: json['available_beds'] ?? 0,
      icuBeds: json['icu_beds'] ?? 0,
      emergencyContact: json['emergency_contact'] ?? 'N/A',
      phoneNumber: json['phone_number'],
      email: json['email'],
      website: json['website'],
      googleMapsLink: json['google_maps_link'],
      rating: (json['rating'] ?? 0.0).toDouble(),
      distance: json['distance']?.toDouble(),
      image: json['image'],
    );
  }
}
